core/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -376,12 +376,17 @@ public static void writeRawField(String field, BytesReference source, XContentBu
         }
     }
 
+    public static BytesReference toXContent(ToXContent toXContent, XContentType xContentType) throws IOException {
+        return toXContent(toXContent, xContentType, false);
+    }
+
     /**
      * Returns the bytes that represent the XContent output of the provided {@link ToXContent} object, using the provided
      * {@link XContentType}. Wraps the output into a new anonymous object.
      */
-    public static BytesReference toXContent(ToXContent toXContent, XContentType xContentType) throws IOException {
+    public static BytesReference toXContent(ToXContent toXContent, XContentType xContentType, boolean humanReadable) throws IOException {
```

Comment:
nice, this method could be potentially called in all our tests so that we randomize the human readable flag, just to make sure we don't miss any of those additional fields. I think we haven't and this is the first time we encounter this but tests are the only way to actually verify. Maybe we can do this as a followup?